ogr_fdw.c

```diff
@@ -1265,3 +1283,286 @@ ogrEndForeignScan(ForeignScanState *node)
 
 
 #endif /* PostgreSQL 9.3 version check */
+
+#if PG_VERSION_NUM >= 90500
+/*
+ * PostgreSQL 9.5 or above.  Import a foreign schema
+ */
+static List *
+ogrImportForeignSchema(ImportForeignSchemaStmt *stmt, Oid serverOid)
+{
+	List	   *commands = NIL;
+	ForeignServer *server;
+	
+	List *options;
+	ListCell *cell;
+	char *sGeomType;
+	bool check_schema = false;
+	bool		launder_column_names = true;
+	bool		launder_table_names = true;	
+	StringInfoData buf;
+	OgrConnection ogr;
+	int i;
+	int j;
+	int k;
+	char layer_name[STR_MAX_LEN];
+	char table_name[STR_MAX_LEN];
+	ListCell   *lc;
+	bool include_item = false;
+	OGRDataSourceH ogr_ds = NULL;
+	OGRSFDriverH ogr_dr = NULL;
+	OGRFeatureDefnH ogr_fd = NULL;
+	OGRLayerH ogr_lyr = NULL;
+
+	/** check table prefix if remote_schema asked for is not ogr_all **/
+	check_schema = !( strcmp(stmt->remote_schema, "ogr_all") == 0 );
+	
+	elog(NOTICE, "Check schema %d %s", check_schema, stmt->remote_schema);
+	if ( GEOMETRYOID == BYTEAOID){ /* postgis is not in search path */
+		sGeomType = "bytea";	
+	}
+	else {
+		sGeomType = "geometry";	
```

Comment:
I recall explicitly forcing the table structure to match e.g. like a shape file proj and geometry type, but the data still came back with srid 0 and would fail typmod.  So you'd need to change the logic in the query too for that to work no?